mne/io/eeglab/eeglab.py

```diff
@@ -260,6 +330,272 @@ def read_epochs_eeglab(input_fname, events=None, event_id=None, montage=None,
     return epochs
 
 
+def _bunchify(mapping, name='BU'):
+    """Convert mappings to Bunches recursively.
+
+    Based on https://gist.github.com/hangtwenty/5960435.
+    """
+    if isinstance(mapping, Mapping):
+        for key, value in list(mapping.items()):
+            mapping[key] = _bunchify(value)
+        return _bunch_wrapper(name, **mapping)
+    elif isinstance(mapping, list):
+        return [_bunchify(item) for item in mapping]
+    return mapping
+
+
+def _bunch_wrapper(name, **kwargs):
+    """Convert mappings to Bunches."""
+    return Bunch(**kwargs)
+
+
+def _bunch_data_2_strs(bunch_data, field, lower=True):
+    """ Takes string values stored as ascii values in numpy arrays
+        in bunch objects and returns list of human-readable strings"""
+
+    str_list = [''.join([chr(x) for x in curr_label.__dict__[field]]).strip()
+                for curr_label in bunch_data]
+    return str_list
+
+
+def _bunch_str_conversions(bunch_data, str_conversion_fields):
+    """ Converts selected fields from bunch object from 1D array
+        aof ascii values to strs"""
+
+    for curr_field in str_conversion_fields:
+        c1 = (len(bunch_data) > 0)
+        c2 = (c1 and (curr_field in bunch_data[0]))
+        c3 = (c2 and
+              not isinstance(bunch_data[0].__dict__[curr_field], str))
+        if c3:
+            str_data = _bunch_data_2_strs(bunch_data, curr_field)
+            for ctr, curr_str in enumerate(str_data):
+                bunch_data[ctr].__dict__[curr_field] = curr_str
+    return bunch_data
+
+
+def _bunch_derefs(orig, bunch_data, deref_fields):
+    """ Dereferences h5py.h5r.Reference objects. Ensures that each
+        field of bunch object with dereferenced objects stores them
+        in a list, even if that list has only 1 element. """
+    try:
+        import h5py
+    except ImportError:
+        raise RuntimeError('Reading v7+ MATLAB format .set',
+                           'requires h5py, which could not',
+                           'be imported')
```

Comment:
This check several places, it should be possible to do it only one place I think